### PR TITLE
refactor: remove count from spare share splitter

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/873
Closes https://github.com/informalsystems/audit-celestia/issues/20

### CODE REVIEW

File: pkg/shares/sparse_shares_test.go [hunk 1]
```diff
@@ -126,47 +126,86 @@ func TestParsePaddedMsg(t *testing.T) {
...
-	sss := NewSparseShareSplitter()
-	smallMsg := generateRandomMessage(appconsts.SparseShareContentSize / 2)
-	sss.Write(smallMsg)
-
-	shares := sss.Export()
-
-	got := shares[0][appconsts.NamespaceSize : appconsts.NamespaceSize+appconsts.ShareInfoBytes][0]
-
-	isSequenceStart := true
-	want, err := NewInfoByte(appconsts.ShareVersion, isSequenceStart)
```
Comment (rootulp) [inline pkg/shares/sparse_shares_test.go:150]: Apologies for including a refactor to the info byte tests while implementing `TestSparseShareSplitterCount`. The refactor can be extracted to a separate PR if requested.

**Review by adlerjohn — APPROVED** [review]
utACK
